src/tnc.rs

```diff
@@ -152,9 +155,8 @@ trait TncImpl: Send + Sync {
 }
 
 /// A local or remote TNC attached to a radio, which can send and receive frames.
-pub struct Tnc {
-    imp: Box<dyn TncImpl>,
-}
+#[derive(Clone)]
+pub struct Tnc(Arc<Mutex<TncInner>>);
```

Comment:
You know, since all of `TncInner`'s internal state has to be behind `Arc<Mutex<_>>` anyway in order to share subscribers with the worker thread, I think it could just take `&self` for both `incoming()` and `send_frame()` and then we don't need the `Mutex<_>` layer on this line. i.e. just `Arc<TncInner>`.

Reply:
Good point.

Reply:
Looks like we can't remove the `Mutex<_>` since we need to be able to borrow it mutably. `Arc<TncInner>` doesn't allow for mutable borrows since `TncInner` doesn't implement send + sync (aiui).

Reply:
Ah not to worry, an extra layer of lock won't hurt anything